packages/codebytes/src/MonacoEditor/index.tsx

```diff
@@ -4,31 +4,38 @@
 // Monaco as a shared package RFC https://www.notion.so/codecademy/Monaco-editor-as-a-shared-package-1f4484db165b4abc8394c3556451ef6a
 
 import loadable from '@loadable/component';
+import { OnMount } from '@monaco-editor/react';
 import React, { useCallback, useRef } from 'react';
-import MonacoEditor from 'react-monaco-editor';
 import ResizeObserver from 'react-resize-observer';
 
 import { dark } from './theme';
 import { Monaco } from './types';
 
-const ReactMonacoEditor = loadable(() => import('react-monaco-editor'));
+const ReactMonacoEditor = loadable(() => import('@monaco-editor/react'));
 
 export type SimpleMonacoEditorProps = {
   value: string;
   language: string;
   onChange?: (value: string) => void;
 };
 
+type ThemedEditor = Parameters<OnMount>[0];
+
 export const SimpleMonacoEditor: React.FC<SimpleMonacoEditorProps> = ({
   value,
   language,
   onChange,
 }) => {
-  const editorRef = useRef<MonacoEditor>(null);
-  const editorWillMount = useCallback((monaco: Monaco) => {
-    monaco.editor.defineTheme('dark', dark);
-    monaco.editor.setTheme('dark');
-  }, []);
+  /* The library doesn't export the required type for the editor and so we'll use any for now */
+  const editorRef = useRef<any>(null);
```

Comment:
I believe you can get this by importing the editor like this:

`import { editor } from 'monaco-editor/esm/vs/editor/editor.api';`

and then using like this:
```suggestion
  const editorRef = useRef<editor: editor.IStandaloneCodeEditor>(null);
```

Reply:
@aionate0812 I love this idea, and I tried to do something similar by digging a little deeper and assigning the `IStandaloneCodeEditor` type. the only issue I'm running into at present is that typescript starts complaining when i try to assign `editorRef.current`, mentioning that it's read only

`editorRef.current = editor`

I like this idea, I can see if I can play with it, not quite sure at the moment how to bypass the read only error and have functionality be the same

Reply:
actually, nevermind, I think I figured it out just by typing out my issue here :) this helped: https://stackoverflow.com/questions/58017215/what-typescript-type-do-i-use-with-useref-hook-when-setting-current-manually

Reply:
I'm looking at this, and it's getting me closer, but not quite there yet: https://stackoverflow.com/questions/58017215/what-typescript-type-do-i-use-with-useref-hook-when-setting-current-manually

Reply:
Ok, yep! Got it to work